xbmc/addons/AddonManager.cpp

```diff
@@ -320,6 +320,24 @@ bool CAddonMgr::Init()
 
   FindAddons();
 
+  // disable some system addons by default because they are optional
+  VECADDONS addons;
+  GetAddons(ADDON_PVRDLL, addons);
+  GetAddons(ADDON_AUDIODECODER, addons);
+  GetAddons(ADDON_AUDIOENCODER, addons);
+  std::string systemAddonsPath = CSpecialProtocol::TranslatePath("special://xbmc/addons");
+  for (size_t i = 0; i<addons.size(); i++)
+  {
```

Comment:
unless I'm mistaken, the reason for using an iterator is that it's often faster code when not dealing with a vector, not because of the style, because style-wise all this c++ templating is not very nice (`std::vector<foo>::const_iterator`)

in this situation, with vector access being linear, you're probably not going to see any speed improvements, and instead of the index var you now have an iterator.

still, for consistency, I'd use the iterator instead here.